invenio_rdm_migrator/actions/base.py

```diff
@@ -105,3 +87,37 @@ def prepare(self, **kwargs):
         # resolve entry references
         self._resolve_references()
         yield from self._generate_rows(**kwargs)
+
+
+class TransformAction(Action, ABC):
+    """Transform action.
+
+    Detects/matches an action type based on transaction data and transforms the data
+    into a target ``LoadAction``.
+    """
+
+    load_cls: ClassVar[type[LoadAction]] = None
+
+    def __init__(self, tx):
+        """Constructor."""
+        assert self.load_cls is not None
+        self.tx = tx
+        super().__init__()
+
+    @property
+    def data_cls(self):
+        """Load data class."""
+        return self.load_cls.data_cls
+
+    def transform(self):
+        """Transforms an action."""
+        return self.load_cls(self._transform_data())
+
+    @classmethod
+    def matches_action(cls, tx):  # pragma: no cover
```

Comment:
`match`? or `matches`?  since it is already an `Action` class, is a bit redundant?

Reply:
:+1: it could also be just `TransformAction.matches(cls, tx)`  (reads well in English too)